music_assistant/providers/podcast-index/provider.py

```diff
@@ -0,0 +1,636 @@
+        """
+        if media_type != MediaType.PODCAST_EPISODE:
+            raise MediaNotFoundError("Stream details only available for episodes")
+
+        try:
+            podcast_id, episode_id = item_id.split("|", 1)
+
+            # Use direct episode lookup for efficiency
+            response = await self._api_request("episodes/byid", params={"id": episode_id})
+            episode_data = response.get("episode")
+
+            if episode_data:
+                stream_url = episode_data.get("enclosureUrl")
+                if stream_url:
+                    return StreamDetails(
+                        provider=self.lookup_key,
+                        item_id=item_id,
+                        audio_format=AudioFormat(
+                            content_type=ContentType.try_parse(
+                                episode_data.get("enclosureType") or "audio/mpeg"
+                            ),
+                        ),
+                        media_type=MediaType.PODCAST_EPISODE,
+                        stream_type=StreamType.HTTP,
+                        path=stream_url,
+                        allow_seek=True,
+                    )
+
+        except (ProviderUnavailableError, InvalidDataError):
+            # Re-raise these specific errors
+            raise
+        except ValueError as err:
+            # Handle malformed episode ID
+            raise InvalidDataError(f"Invalid episode ID format: {item_id}") from err
+        except Exception as err:
+            self.logger.warning("Unexpected error getting stream for %s: %s", item_id, err)
+
+        raise MediaNotFoundError(f"Stream not found for {item_id}")
+
+    async def get_item(self, media_type: MediaType, prov_item_id: str) -> Podcast | PodcastEpisode:
+        """Get single MediaItem from provider."""
+        if media_type == MediaType.PODCAST:
+            return await self.get_podcast(prov_item_id)
+        elif media_type == MediaType.PODCAST_EPISODE:
+            return await self.get_podcast_episode(prov_item_id)
+        else:
+            raise MediaNotFoundError(f"Media type {media_type} not supported by this provider")
+
+    async def _fetch_podcasts(
+        self, endpoint: str, params: dict[str, Any] | None = None
+    ) -> list[Podcast]:
+        """Fetch and parse podcasts from API endpoint."""
+        response = await self._api_request(endpoint, params)
+        podcasts = []
+        for feed_data in response.get("feeds", []):
+            podcast = parse_podcast_from_feed(
+                feed_data, self.lookup_key, self.domain, self.instance_id
+            )
+            if podcast:
+                podcasts.append(podcast)
+        return podcasts
+
+    async def _api_request(
+        self, endpoint: str, params: dict[str, Any] | None = None
+    ) -> dict[str, Any]:
+        """Make authenticated request to Podcast Index API."""
+        self.logger.debug("Making API request to %s with params: %s", endpoint, params)
```

Comment:
```suggestion
        self.logger.log(LOG_LEVEL_VERBOSE, "Making API request to %s with params: %s", endpoint, params)
```

This might be more something for verbose logging. You can import that constant from the main constants file